builder/azure/arm/template_factory.go

```diff
@@ -156,9 +156,16 @@ func GetVirtualMachineDeployment(config *Config) (*resources.Deployment, error)
 		}
 	}
 
+	if config.BuildDiskEncryptionSetId != "" {
+		err = builder.SetDiskEncryptionSetID(config.BuildDiskEncryptionSetId)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	if len(config.AdditionalDiskSize) > 0 {
-		isManaged := config.CustomManagedImageName != "" || (config.ManagedImageName != "" && config.ImagePublisher != "") || config.SharedGallery.Subscription != ""
-		err = builder.SetAdditionalDisks(config.AdditionalDiskSize, config.tmpDataDiskName, isManaged, config.diskCachingType)
+		isLegacyVHD := config.CustomManagedImageName == "" && config.ManagedImageName == "" && config.ImagePublisher == "" && config.SharedGalleryDestination.SigDestinationGalleryName == ""
```

Comment:
"isManaged" is a bit confusing here, since we want to set the ManagedDisk properties even if just directly publishing to SIG, this was setup to check the SIG subscription, but it should be set to the SIG Destination gallery name, as you don't have to set a subscription if you just inherit it from higher up in the template